pkg/util/voljournal.go

```diff
@@ -109,46 +118,54 @@ type CSIJournal struct {
 	// Ceph volume was created
 	csiNameKey string
 
+	// CSI image-name key in per Ceph volume object map, containing RBD image-name
+	// of this Ceph volume
+	csiImageKey string
+
 	// source volume name key in per Ceph snapshot object map, containing Ceph source volume uuid
 	// for which the snapshot was created
 	cephSnapSourceKey string
 
-	// volume name prefix for naming on Ceph rbd or FS, suffix is a uuid generated per volume
-	namingPrefix string
-
 	// namespace in which the RADOS objects are stored, default is no namespace
 	namespace string
 }
 
-// CSIVolumeJournal returns an instance of volume keys
+// NewCSIVolumeJournal returns an instance of CSIJournal for volumes
 func NewCSIVolumeJournal() *CSIJournal {
 	return &CSIJournal{
 		csiDirectory:            "csi.volumes",
 		csiNameKeyPrefix:        "csi.volume.",
 		cephUUIDDirectoryPrefix: "csi.volume.",
 		csiNameKey:              "csi.volname",
-		namingPrefix:            "csi-vol-",
+		csiImageKey:             "csi.imagename",
 		cephSnapSourceKey:       "",
 		namespace:               "",
 	}
 }
 
-// CSISnapshotSnapshot returns an instance of snapshot keys
+// NewCSISnapshotJournal returns an instance of CSIJournal for snapshots
 func NewCSISnapshotJournal() *CSIJournal {
 	return &CSIJournal{
 		csiDirectory:            "csi.snaps",
 		csiNameKeyPrefix:        "csi.snap.",
 		cephUUIDDirectoryPrefix: "csi.snap.",
 		csiNameKey:              "csi.snapname",
-		namingPrefix:            "csi-snap-",
+		csiImageKey:             "csi.imagename",
```

Comment:
csi.imagename or snapname?

Reply:
Doesn't really matter, it's an arbitrary name. I'm fine changing it to `csi.snapname`

Reply:
While it is an arbitrary key name, I think it shouldn't conflict with either `csi.volname` or `csi.snapname` which are already the `csiNameKey`'s. This key specifically points to the Ceph image involved, so I think in both cases `csi.imagename` applies.